### PR TITLE
Update Documentation BBE with documentation reference examples

### PR DESCRIPTION
**Pull request by dulvinw**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/documentation/documentation.bal [hunk 1]
```diff
@@ -30,10 +31,21 @@ public type DummyObject abstract object {
...
     string fieldOne;
     int fieldTwo;
 };
+
+// This is the documentation attachment of the `dummyFunction`.
+# The function `dummyFunction` is a user-defined function with
+# + x - One input parameter.
+# + y - Another input parameter.
+# + return - the sum of input parameters.
+# This function add parameter `x` and parameter `y`.
```
Comment (MaryamZi) [inline examples/documentation/documentation.bal]: For consistency, can we update this block as 

```ballerina
# This `dummyFunction` function accepts two `int` parameters and returns the sum of the two.
#
# + x - First parameter.
# + y - Second parameter.
# + return - The sum of the two parameters.
```

File: examples/documentation/documentation.bal [hunk 2]
```diff
@@ -30,10 +31,21 @@ public type DummyObject abstract object {
...
+public function dummyFunction(int x, int y) returns int {
+    return x+y;
```
Comment (MaryamZi) [inline examples/documentation/documentation.bal]:
```suggestion
    return x + y;
```

File: examples/documentation/documentation.bal [hunk 3]
```diff
@@ -30,10 +31,21 @@ public type DummyObject abstract object {
 // This is the documentation attachment of the `DummyRecord` record.
 # `DummyRecord` is a user-defined record.
 # This `DummyRecord` has a `string` data field and an `int` data field.
+# This is a reference to a function `dummyFunction`.
```
[the remark this answers is not in the document]
Comment (dulvinw) [reply examples/documentation/documentation.bal]: This is to show that we can refer to top level nodes in BFM. Is it not needed in this BBE? WDYT?

File: examples/documentation/documentation.bal (hunk 1 above)
Comment (dulvinw) [reply examples/documentation/documentation.bal]: Done.

File: examples/documentation/documentation.bal [hunk 4]
```diff
@@ -37,3 +37,11 @@ public type DummyRecord record {
     string fieldOne;
     int fieldTwo;
 };
+
+// This is the documentation attachment of the `dummyFunction` function.
+# This function returns the fieldTwo of type `DummyRecord`.
```
Comment (MaryamZi) [inline examples/documentation/documentation.bal]: Shall we add an empty line between this and the param/return?

```ballerina
// This is the documentation attachment of the `dummyFunction` function.
# This function returns the fieldTwo of type `DummyRecord`.
#
# + recordValue - Parameter of type `DummyRecord`.
# + return - Return the fieldTwo of type `DummyRecord`.
```

Comment (MaryamZi) [inline examples/documentation/documentation.bal]:
```suggestion
# This function returns the `fieldTwo` field of the `DummyRecord`-typed 
# record value passed as an argument.
```

File: examples/documentation/documentation.bal [hunk 5]
```diff
@@ -37,3 +37,11 @@ public type DummyRecord record {
...
+# + recordValue - Parameter of type `DummyRecord`.
+# + return - Return the fieldTwo of type `DummyRecord`.
```
Comment (MaryamZi) [inline examples/documentation/documentation.bal]:
```suggestion
# + return - The `fieldTwo` field of the record value passed as an argument.
```

File: examples/documentation/documentation.bal (hunk 4 above)
Comment (dulvinw) [reply examples/documentation/documentation.bal]: Done.

**Pull request merged**